model/jpa/src/main/java/org/keycloak/models/jpa/JpaRealmProvider.java

```diff
@@ -304,7 +289,7 @@ public boolean removeRole(RealmModel realm, RoleModel role) {
         RoleEntity roleEntity = em.getReference(RoleEntity.class, role.getId());
         String compositeRoleTable = JpaUtils.getTableNameForNativeQuery("COMPOSITE_ROLE", em);
         em.createNativeQuery("delete from " + compositeRoleTable + " where CHILD_ROLE = :role").setParameter("role", roleEntity).executeUpdate();
-        realm.getClients().forEach(c -> c.deleteScopeMapping(role));
+        realm.getClients(0, realm.getClientsCount().intValue()).forEach(c -> c.deleteScopeMapping(role));
```

Comment:
Same comment as earlier. We need a realm.getClients() that can just call realm.getClients(null,null) without pagination instead of fetching count first.